.github/workflows/test.yml

```diff
@@ -80,6 +80,10 @@ jobs:
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
           fetch-depth: 0
+      - name: Install ZLIB
+        shell: cmd
+        run: |
+          powershell -Command "(Invoke-WebRequest -Uri https://git.io/JnHTY -OutFile install_zlib.bat)"; ./install_zlib.bat; del install_zlib.bat
```

Comment:
Is this script reliable?

Basically I did the following steps to install the zlib library on Windows:

```
git clone https://github.com/madler/zlib.git
cd zlib
cmake -S . -B build -DCMAKE_INSTALL_PREFIX=/tmp/zlib -DZLIB_BUILD_EXAMPLES=OFF
cmake --build build --config Release
cmake --install build
```

Reply:
I'm not sure if it is reliable, this is just a away I found that can walk around the zlib requirement. I will try your way later.

Reply:
What about using `vcpkg`? I believe we may need to add more libraries to the Windows CI. Using `vcpkg` may be easier. Below is the suggestion from deepseek.

```
name: CI

on: [push]

jobs:
  build:
    runs-on: windows-latest

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Set up vcpkg
      run: |
        git clone https://github.com/microsoft/vcpkg.git
        .\vcpkg\bootstrap-vcpkg.bat

    - name: Install zlib
      run: .\vcpkg\vcpkg install zlib:x64-windows

    - name: Build your project
      run: |
        mkdir build
        cd build
        cmake .. -DCMAKE_TOOLCHAIN_FILE=../vcpkg/scripts/buildsystems/vcpkg.cmake
        cmake --build .
```

Reply:
Let's go this way, good news is that github action windows runner has vcpkg installed [1]. I also changed the windows runner image to Windows2022, which resolved the `fatal error C1001: Internal compiler error`. We are using the newest ubuntu runner version gh-actions supported, so I guess we can use a slight newer version of windows(the newest is Windows2025).


[1] https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md